### ISSUE TITLE
Routes for User Login

### ISSUE THREAD

**Issue by alltrust** [alltrust/B-Article-Scraper#2]
## Summary
> Create login route for user to be able to login with already registered and stored credentials.

## Expected System Behavior
> User will login with their email or username and password. If user has previously registered, user will be able to login. If user credentials are inaccurate or unregistered user can't login, but can register.
 
## Acceptance Criteria

-  use bcrypt to match input password and saved password
-  error handling for mismatched password and incorrect or unregistered email/username combination. 
-  password reset option 
-  send user credentials to client

**Comment by sebastianserrano** [comment]
Why not just use either username or email to login? Seems redundant to have both options and confusing for new users

**Comment by alltrust** [comment]
You are right, that would be redundant. My intentions are to actually to use either/or.

**Comment by alltrust** [comment]
- Username or email is now used to configure whether the user can login with their preferred credential and password. 
- error handling is set up as well with an error handling middleware. The error middleware catches all thrown errors that are sent from the controllers.
